support `Oscar.is_unicode_allowed` in matrix display

ThomasBreuer opened the pull request:
- provide ASCII and unicode variants of `labelled_matrix_formatted`,
  depending on `Oscar.is_unicode_allowed()`
- added tests for the two variants
- mark the doctests in question such that the unicode variant is used
  (in order to show this variant in the manual)

ThomasBreuer committed:
support `Oscar.is_unicode_allowed` in matrix display

- provide ASCII and unicode variants of `labelled_matrix_formatted`,
  depending on `Oscar.is_unicode_allowed()`
- added tests for the two variants
- mark the doctests in question such that the unicode variant is used
  (in order to show this variant in the manual)

fingolfin commented on src/Groups/MatrixDisplay.jl:
````diff
@@ -108,7 +118,7 @@ The following attributes of `io` are supported.
 
 ## Examples
-```jldoctest
+```jldoctest; setup = :(Oscar.allow_unicode(true))
````
Will this turn off the unicode support later on again? If not, couldn't this affect other examples?

ThomasBreuer replied:
In fact I had expected all kinds of technical problems, the current test failures for Julia nightly are perhaps some of them.
(The failures happen in `printing.jl`, the first test file that gets evaluated.)
I am now also checking the behaviour of the `setup` code w.r.t. different doctests

ThomasBreuer replied:
@fingolfin I have changed the approach: Do not set globals in `setup` code of jldoctest.

Thus setting/resetting the `is_unicode_allowed` flag is now visible in the docstring examples.

Still it might be a bad idea to show unicode in the manual, whereas it is switched off in the session by default.
On the other hand, unicode looks just nicer, in particular in the manual.
(Perhaps we will eventually need a general decision that unicode is banned also from the manual.)

fingolfin reviewed (approved):
Nice

ThomasBreuer committed:
do not set globals in `setup` code of `jldoctest`

Thus setting/resetting the `is_unicode_allowed` flag is now visible
in the docstring examples.

Still it might be a bad idea to show unicode in the manual,
whereas it is switched off in the session by default.
On the other hand, unicode looks just nicer, in particular in the manual.

thofma commented:
I wonder if we should have
```
with_unicode() do
  blabla
end
```
syntax to make it easier to test it (which takes care of setting/resetting the setting).

ThomasBreuer committed:
introduce `with_unicode`

fingolfin commented:
@thofma so @ThomasBreuer introduced `with_unicode`, can you please look if this is how you had it in mind?

Other than that, this PR is IMHO read for merging

ThomasBreuer commented:
@thofma Using `with_unicode() do` syntax is fine in test files. For doctests, the explicit `show` or `print` calls are a bit annoying.

thofma commented:
Yes to Max and yes to Thomas, I agree that this does not look pretty. Any suggestion?

For doctests in the markdown files in the `test/*.md` files, one could reset the setup as in https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#DocTestSetup-in-@meta-blocks, but I don't think this works for the docstrings themselves.

ThomasBreuer commented:
The available (recommended?) [`setup` code](https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#Block-level-setup-code) for docstrings has the disadvantage that one cannot reset the global variable after the docstring. I also think that such invisible magic would be not desirable for the manual because users will not understand why their session looks different to the documentation.

ThomasBreuer committed:
try to simplify the use of `with_unicode`

ThomasBreuer commented:
@thofma  @fingolfin What about the changed behaviour of `with_unicode` proposed in commit b993230 ?

thofma commented:
I don't think this does what we want, since it makes the block just return the result of `f()`. So it is returned from the block and then it is printed I believe.

ThomasBreuer commented:
@thofma You are right. I was misled by the (for me) unusual way how preferences behave. It is quite dangerous to set a preference not just for the current Julia session but for each session later on.

ThomasBreuer committed:
Revert "try to simplify the use of `with_unicode`"

This reverts commit b993230b788a6b2796b1c224696f08fff75b4187.
